akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -78,8 +77,31 @@ private[akka] final class InterceptorImpl[O, I](val interceptor: BehaviorInterce
   }
 
   override def receiveSignal(ctx: typed.TypedActorContext[O], signal: Signal): Behavior[O] = {
-    val interceptedResult = interceptor.aroundSignal(ctx, signal, signalTarget)
-    deduplicate(interceptedResult, ctx)
+    if (signal == PostStop) {
+      nestedBehavior match {
+        case s: StoppedBehavior[O] ⇒
+          s.postStop match {
+            case OptionVal.Some(callback) ⇒
+              interceptor.aroundSignal(ctx, PostStop, new SignalTarget[I] {
+                def apply(ctx: TypedActorContext[_], signal: Signal): Behavior[I] = {
+                  callback match {
+                    case x: ExtensibleBehavior[I] ⇒
+                      x.receiveSignal(ctx.asInstanceOf[TypedActorContext[I]], PostStop)
+                  }
+                }
+              })
+            case OptionVal.None ⇒
+              interceptor.aroundSignal(ctx, PostStop, new SignalTarget[I] {
+                def apply(ctx: TypedActorContext[_], signal: Signal): Behavior[I] = Behavior.same
+              })
+          }
+      }
+      Behavior.same
```

Comment:
This is messy. I tried making `PostStop` an `ExtensibleBehavior` instead and rely on less specific logic here, but I didn't get it to work.

Reply:
indeed messy